### ISSUE TITLE
GDALDataFinder no longer finds data path in fiona 1.8.9 with GDAL 3

### ISSUE THREAD

**Issue by jorisvandenbossche** [Toblerity/Fiona#818]
In a new test environment with latest fiona:

```
(test-fiona-189) $ gdal-config --datadir
/home/joris/miniconda3/envs/test-fiona-189/share/gdal
(test-fiona-189) $ ipython
Python 3.7.3 | packaged by conda-forge | (default, Jul  1 2019, 21:52:21) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.9.0 -- An enhanced Interactive Python. Type '?' for help.
In [1]: import fiona 

In [2]: fiona.env.GDALDataFinder().search() 

In [3]: fiona.__version__ 
Out[3]: '1.8.9.post2'
```

while with previous versions (using GDAL 2):

```
In [22]: import fiona  

In [23]: fiona.__version__   
Out[23]: '1.8.6'

In [24]: fiona.env.GDALDataFinder().search()
Out[24]: '/home/joris/miniconda3/envs/dev/share/gdal'
```

From a quick look, I think this is due to 

https://github.com/Toblerity/Fiona/blob/d6ec15f1c2e7a860bc57d35a51e36aeb48377a1a/fiona/_env.pyx#L303

where there is an explicit check that the pcs.csv exists, but this file is removed in GDAL 3 (https://github.com/OSGeo/gdal/pull/1185/files?file-filters%5B%5D=.csv)

**Comment by jorisvandenbossche** [comment]
What I don't fully understand is that this doesn't fail any tests, so is this data path not that important? Or it doesn't get tested in a way to trigger this (eg from installed wheel or conda package)

**Comment by sgillies** [comment]
@jorisvandenbossche there is indeed some unfinished work here. Fiona 1.8.9 will work with GDAL 3 and PROJ 6 if you install fiona from a source distribution. For conda (and PyPI wheels), there are several things yet to do. We need to change the GDAL data file we're looking for as in rasterio: https://github.com/mapbox/rasterio/blob/master/rasterio/_env.pyx#L350. We also need to switch over to GDAL's new API for finding PROJ data. I'll take care of these right away. Testing this is awkward: to do it properly we need to add different build configurations to our already large matrix.

**Comment by jorisvandenbossche** [comment]
> Testing this is awkward: to do it properly we need to add different build configurations to our already large matrix.

Yeah, I don't think writing the script the have a build entry that builds and installs a wheel and runs the tests for that installed version should be very hard, but fitting it into the current travis configuration seems more tricky.

**Comment by sgillies** [comment]
This will be fixed by a4de49b.
